src/transit.js

```diff
@@ -905,12 +905,13 @@ class Transit {
 								`=> Send stream chunk ${requestID}to ${nodeName} node. Seq: ${copy.seq}`
 							);
 
-							this.publish(new Packet(P.PACKET_REQUEST, ctx.nodeID, copy)).catch(
```

Comment:
I think, it would be much easier to add `await` to this row instead of changing the `for` iteration to a map and Promise.all

Reply:
Totally agree. I used to think that you want to send packets concurrently. if don't want to do this, it would be easier to add await in the for iteration

Reply:
Oh, I remembered, this asynchronous iteration is in the event callback function. I'm afraid this is why you didn't add await before.

However, we can use the reduce method to make the publish process execute sequentially.

Reply:
The original code sent all chunks in parallel, but I thought that your original problem was the parallelism or not?

Reply:
No, The original issue was that the back pressure mechanism of the stream was not being correctly utilized, which caused memory leaks under certain conditions. 
In the original code, the stream was resumed before waiting for all the chunks to be fully published. As a result, the consumption progress of the stream did not match the publishing progress of the chunks. This caused the program to buffer excessive data in memory in some cases.